pig/src/main/java/org/apache/zeppelin/pig/PigInterpreter.java

```diff
@@ -0,0 +1,137 @@
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.pig.PigServer;
+import org.apache.pig.impl.logicalLayer.FrontendException;
+import org.apache.pig.tools.pigstats.*;
+import org.apache.zeppelin.interpreter.InterpreterContext;
+import org.apache.zeppelin.interpreter.InterpreterResult;
+import org.apache.zeppelin.interpreter.InterpreterResult.Code;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.*;
+
+/**
+ * Pig interpreter for Zeppelin.
+ */
+public class PigInterpreter extends BasePigInterpreter {
+  private static Logger LOGGER = LoggerFactory.getLogger(PigInterpreter.class);
+
+  private PigServer pigServer;
+  private boolean includeJobStats = false;
+
+  public PigInterpreter(Properties property) {
+    super(property);
+  }
+
+  @Override
+  public void open() {
+    String execType = getProperty("zeppelin.pig.execType");
+    if (execType == null) {
+      execType = "mapreduce";
+    }
+    String includeJobStats = getProperty("zeppelin.pig.includeJobStats");
+    if (includeJobStats != null) {
+      this.includeJobStats = Boolean.parseBoolean(includeJobStats);
+    }
+    try {
+      pigServer = new PigServer(execType);
+    } catch (IOException e) {
+      throw new RuntimeException("Fail to launch PigServer", e);
+    }
+  }
+
+  @Override
+  public void close() {
+    pigServer = null;
+  }
+
+
+  @Override
+  public InterpreterResult interpret(String cmd, InterpreterContext contextInterpreter) {
+    // remember the origial stdout, because we will redirect stdout to capture
+    // the pig dump output.
+    PrintStream originalStdOut = System.out;
+    ByteArrayOutputStream bytesOutput = new ByteArrayOutputStream();
+    File tmpFile = null;
+    try {
+      tmpFile = PigUtils.createTempPigScript(cmd);
+      System.setOut(new PrintStream(bytesOutput));
+      // each thread should its own ScriptState & PigStats
+      ScriptState.start(pigServer.getPigContext().getExecutionEngine().instantiateScriptState());
+      // reset PigStats, otherwise you may get the PigStats of last job in the same thread
+      // because PigStats is ThreadLocal variable
+      PigStats.start(pigServer.getPigContext().getExecutionEngine().instantiatePigStats());
+      PigScriptListener scriptListener = new PigScriptListener();
+      ScriptState.get().registerListener(scriptListener);
+      listenerMap.put(contextInterpreter.getParagraphId(), scriptListener);
+      pigServer.registerScript(tmpFile.getAbsolutePath());
+    } catch (IOException e) {
+      if (e instanceof FrontendException) {
+        FrontendException fe = (FrontendException) e;
+        if (!fe.getMessage().contains("Backend error :")) {
+          // If the error message contains "Backend error :", that means the exception is from
+          // backend.
+          return new InterpreterResult(Code.ERROR, ExceptionUtils.getStackTrace(e));
```

Comment:
I don't feel strongly about it but there has been numerous request to hide exception stack in interpreter results in other interpreters and we generally do no put exception stack in interpreter results.